client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpResponse.java

```diff
@@ -6,43 +6,75 @@
 
 package com.microsoft.rest.v2.http;
 
+import rx.Single;
+
 import java.io.IOException;
 import java.io.InputStream;
 
 /**
  * This class contains all of the details necessary for reacting to a HTTP response from a
  * HttpResponse.
  */
-public interface HttpResponse {
+public abstract class HttpResponse {
     /**
      * Get whether or not this response object has a body.
      * @return Whether or not this response object has a body.
      */
-    boolean hasBody();
+    public abstract boolean hasBody();
 
     /**
      * Get this response object's body as an InputStream. If this response object doesn't have a
      * body, then null will be returned.
      * @return This response object's body as an InputStream. If this response object doesn't have a
      * body, then null will be returned.
      */
-    InputStream bodyAsInputStream();
+    public InputStream bodyAsInputStream() {
+        return bodyAsInputStreamAsync().toBlocking().value();
+    }
+
+    /**
+     * Get this response object's body as an InputStream. If this response object doesn't have a
+     * body, then null will be returned.
+     * @return This response object's body as an InputStream. If this response object doesn't have a
+     * body, then null will be returned.
+     */
+    public abstract Single<? extends InputStream> bodyAsInputStreamAsync();
+
+    /**
+     * Get this response object's body as a byte[]. If this response object doesn't have a body,
+     * then null will be returned.
+     * @return This response object's body as a byte[]. If this response object doesn't have a body,
+     * then null will be returned.
+     * @throws IOException On network error.
+     */
+    public byte[] bodyAsByteArray() throws IOException {
+        return bodyAsByteArrayAsync().toBlocking().value();
+    }
 
     /**
      * Get this response object's body as a byte[]. If this response object doesn't have a body,
      * then null will be returned.
      * @return This response object's body as a byte[]. If this response object doesn't have a body,
      * then null will be returned.
-     * @throws IOException On network issues.
      */
-    byte[] bodyAsByteArray() throws IOException;
+    public abstract Single<byte[]> bodyAsByteArrayAsync();
```

Comment:
yes, if the response is chunked netty can return multiple `HttpContent`s, in that case this should be an `Observable<byte[]>`. If we insert the [HttpObjectAggregator](https://netty.io/4.0/api/io/netty/handler/codec/http/HttpObjectAggregator.html) handler in the pipeline then these chunks will be merged into a single `HttpContent`.